src/OrbitBase/ThreadUtilsLinux.cpp

```diff
@@ -22,12 +22,45 @@ namespace orbit_base {
 // including the terminating null byte ('\0')".
 static constexpr size_t kMaxThreadNameLength = 16;
 
-pid_t GetCurrentThreadId() {
+static constexpr int32_t kInvalidLinuxThreadId = -1;
+static constexpr int32_t kInvalidLinuxProcessId = -1;
+
+static constexpr uint32_t kIntMax = static_cast<uint32_t>(INT_MAX);
+
+uint32_t GetCurrentThreadId() { return GetThreadIdFromNative(GetCurrentThreadIdNative()); }
+
+uint32_t GetCurrentProcessId() { return GetProcessIdFromNative(GetCurrentProcessIdNative()); }
+
+bool IsValidThreadId(uint32_t tid) { return tid <= kIntMax && tid != kInvalidThreadId; }
+
+bool IsValidProcessId(uint32_t pid) { return pid <= kIntMax && pid != kInvalidProcessId; }
+
+pid_t GetCurrentThreadIdNative() {
   thread_local pid_t current_tid = syscall(__NR_gettid);
   return current_tid;
 }
 
-std::string GetThreadName(pid_t tid) {
+pid_t GetCurrentProcessIdNative() { return getpid(); }
+
+uint32_t GetThreadIdFromNative(pid_t tid) {
+  return tid >= 0 ? static_cast<uint32_t>(tid) : orbit_base::kInvalidThreadId;
```

Comment:
I think invalid input should lead to checkfail...

CHECK(tid == -1 || tid >= 0);

and then you can just static_cast.
same thing for GetNative

Reply:
Done, thanks